src/main/java/com/code_intelligence/jazzer/junit/Utils.java

```diff
@@ -127,12 +143,89 @@ static boolean permissivelyParseBoolean(String value) {
   }
 
   /**
-   * Convert the string to ISO 8601 (https://en.wikipedia.org/wiki/ISO_8601#Durations).
-   * We do not allow for duration units longer than hours, so we can always prepend PT.
+   * Convert the string to ISO 8601 (https://en.wikipedia.org/wiki/ISO_8601#Durations). We do not
+   * allow for duration units longer than hours, so we can always prepend PT.
    */
   static long durationStringToSeconds(String duration) {
     String isoDuration =
         "PT" + duration.replace("sec", "s").replace("min", "m").replace("hr", "h").replace(" ", "");
     return Duration.parse(isoDuration).getSeconds();
   }
+
+  /**
+   * Creates {@link Arguments} for a single invocation of a parameterized test that can be
+   * identified as having been created in this way by {@link #isMarkedInvocation}.
+   *
+   * @param displayName the display name to assign to every argument
+   */
+  static Arguments getMarkedArguments(Method method, String displayName) {
+    return arguments(stream(method.getParameterTypes())
+                         .map(Utils::getMarkedInstance)
+                         // Wrap in named as toString may crash on marked instances.
+                         .map(arg -> named(displayName, arg))
+                         .toArray(Object[] ::new));
+  }
+
+  /**
+   * @return {@code true} if and only if the arguments for this test method invocation were created
+   * with {@link #getMarkedArguments}
+   */
+  static boolean isMarkedInvocation(ReflectiveInvocationContext<Method> invocationContext) {
+    if (invocationContext.getArguments().stream().anyMatch(Utils::isMarkedInstance)) {
+      if (invocationContext.getArguments().stream().allMatch(Utils::isMarkedInstance)) {
+        return true;
+      }
+      throw new IllegalStateException(
+          "Some, but not all arguments were marked in invocation of " + invocationContext);
+    } else {
+      return false;
+    }
+  }
+
+  private static final ClassValue<Object> uniqueInstanceCache = new ClassValue<Object>() {
```

Comment:
It's essentially a fixed version of `WeakConcurrentHashMap<Class<?>, T>` if that were to exist - if the GC collects a class, its entry in the `ClassValue` will go away. Getting this right with a regular `Map` is apparently quite tricky since the value will usually hold a reference to the class.

Reply:
Thanks for the clarification! If the value still holds a strong reference to the class or an instance of it, the class can not be unloaded/gced, can it? Regardless, good to know that `ClassValue` exists :smiley: